File: new_75/11_k_m.java
class Solution {
    public int maxArea(int[] height) {
      int maxArea=0,right=height.length-1,left=0;
      while(left<right){
          maxArea=Math.max(maxArea,(right-left)*Math.min(height[right],height[left]));
          if(height[right]<height[left]){
              right--;
          }else left++;
      }
      return maxArea;  
    }
}

File: new_75/11_k_m.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = 0
        right = len(height)-1
        left = 0
        while left<right:
            maxArea = max(maxArea, (right-left)*min(height[right],height[left]))
            if height[right]<height[left]:
                right -= 1
            else:
                left += 1
        return maxArea
